src/lib/utils/redisCache.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-console */
+import { DataCache } from 'next-drupal'
+import { createClient, RedisClientType } from 'redis'
+
+let redisClient: RedisClientType
+let isReady: boolean
+
+async function getCache(): Promise<RedisClientType> {
+  if (!isReady) {
+    // connects to a running redis server, defaults to 127.0.0.1:6379
+    redisClient = createClient({
+      url: process.env.REDIS_URL,
+    })
+    redisClient.on('error', (err) => console.error(`Redis Error: ${err}`))
+    redisClient.on('connect', () => console.info('Redis connected'))
+    redisClient.on('reconnecting', () => console.info('Redis reconnecting'))
+    redisClient.on('ready', () => {
+      isReady = true
+      console.info('Redis ready!')
+    })
+    await redisClient.connect()
+  }
+  return redisClient
+}
+
+getCache()
+  .then((connection) => {
+    redisClient = connection
+  })
+  .catch((err) => {
+    console.error({ err }, 'Failed to connect to Redis')
+  })
```

Comment:
This feels weird to me. I'm guessing this is being called here so as to jumpstart the redis connection rather than wait until the first attempt to set or get, but it feels like that should be explicitly done by the thing that actually uses it (in this case, the DrupalClient object). This sort of top-level code feels like an anti-pattern.

I think I'm envisioning something like this, though I might be missing something:
```
/*redisCache.ts*/
export async function createRedisCache(url): DataCache {
  const redisClient = createClient({ url })
  redisClient.on('error', (err) => console.error(`Redis Error: ${err}`))
  redisClient.on('connect', () => console.info('Redis connected'))
  redisClient.on('reconnecting', () => console.info('Redis reconnecting'))
  redisClient.on('ready', () => {
    isReady = true
    console.info('Redis ready!')
  })
  
  await redisClient.connect()
  
  return {
    async set(key, value) {
      return await redisClient.set(key, value)
    },
    async get(key) {
      return await redisClient.get(key)
    },
  }
}
```
```
/*drupalClient.ts*/
export const drupalClient = new DrupalClient(baseUrl, {
  fetcher,
  useDefaultResourceTypeEntry: true,
  throwJsonApiErrors: false,
  auth: {
    clientId: process.env.DRUPAL_CLIENT_ID,
    clientSecret: process.env.DRUPAL_CLIENT_SECRET,
  },
  cache: createRedisCache(process.env.REDIS_URL),
  previewSecret: process.env.DRUPAL_PREVIEW_SECRET,
  forceIframeSameSiteCookie: process.env.NODE_ENV === 'development',
})
```

Reply:
Though I'm now thinking about the implications of this being async.

Reply:
Though I'm now realizing there are some implications of this being async.